walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,57 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.PriorityQueue;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    int initialQueueCapacity = 1;
+    if (events.size() > 0) {
+      initialQueueCapacity = events.size();
+    }
+    PriorityQueue<TimeRange> blockingIntervals =
+        new PriorityQueue<>(initialQueueCapacity, TimeRange.ORDER_BY_START);
+    for (Event event : events) {
+      if (eventContainsAttendee(event, attendees)) {
+        blockingIntervals.add(event.getWhen());
+      }
+    }
+    ArrayList<TimeRange> freeSlots = new ArrayList<>();
```

Comment:
Not entirely related to this case: take a look at "Item 64: Refer to objects by their interfaces" of Effective Java about why it's usually better to use interfaces :)